File: HealthcarePatientAPIPython/models/person.py
from pydantic import BaseModel, Field, EmailStr
from typing import Optional
from datetime import datetime
from utils.location import Location  # Import Location model

# Person model
class Person(BaseModel):
    first_name: str
    last_name: str
    contact_name: Optional[str]
    mobile_phone: Optional[str]
    fax: Optional[str]
    email: Optional[EmailStr]
    location: Optional[Location]  # Updated to use the Location model
    is_active: bool = True
    date_of_birth: datetime

File: HealthcareUI/Models/Person.cs
using System.ComponentModel;

namespace HealthcareUI.Utils
{

    public class Person
    {
        private DateTime _DateOfBirth;


        public string FirstName { get; set; }

        public string LastName { get; set; }

        public string ContactName { get; set; }


        public string MobilePhone { get; set; }

        public string Fax { get; set; }

        public string Email { get; set; }

        public Location Location { get; set; } = new();

        public bool IsActive
        {
            get; set;
        } = true;
        public DateTime DateOfBirth
        {
            get { return _DateOfBirth; }
            set {

                if (_DateOfBirth.Equals(value))
                    return;

                _DateOfBirth = value;
            }
        }

    }
}
